qt/python/mantidqt/project/encoderfactory.py

```diff
@@ -19,16 +23,12 @@ def find_encoder(cls, obj):
         :param obj: The object for encoding
         :return: Encoder or None; Returns the Encoder of the obj or None.
         """
-        for encoder, compatible_check in cls.encoder_list:
-
-            # Perform check if optional_cb is present if it is, then check whether the the optional_cb returns true for
-            # this object.
-            if compatible_check is not None and compatible_check(obj):
-                return encoder()
-
-            if encoder().has_tag(obj.__class__.__name__):
-                return encoder()
-        return None
+        obj_encoders = [encoder for encoder, compatible in cls.encoder_list if compatible(obj, encoder)]
+        if len(obj_encoders) > 1:
+            raise RuntimeError("EncoderFactory: One or more encoder type claims to work with the passed obj: "
+                               + obj.__class__.__name__)
+        elif len(obj_encoders) == 1:
+            return obj_encoders[0]()
```

Comment:
Should we not raise an error if no encode can be found either? A bit like `dict[]` when a key is missing?

Reply:
Having 0 encoders should also be ok. It is only an error if there is more than one, as many QWidget Object are pushed through this function.

Reply:
When feeding into this method we don't know if there will be an encoder, as anyone could put anything in there, as especially with current usage and implementation we use all top level widgets.

Reply:
Okay. So we could raise a `KeyError` and catch that in the calling code as an alternative but returning `None` is also acceptable. I know Python defaults to `return None`  but can we be explicit with an `else`. A reader will then not think that there is case that has not been thought about.

Reply:
I dislike making people handle errors, returning None explicitly makes the most sense here in my opinion.

Reply:
Callers already have to handle the `RuntimeError` from line 28 but `None` is an acceptable alternative here I think as no encoder is a not deemed a failure path.